autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -53,9 +53,14 @@ def __init__(
                 - Other kwargs: Except verbose, others are passed directly to ConversableAgent.
         """
         # Use AutoGen OpenAIWrapper to create a client
-        oai_wrapper = OpenAIWrapper(**llm_config)
-        if len(oai_wrapper._clients) > 1:
+        model_name = "gpt-4-1106-preview"
+        if len(llm_config["config_list"]) > 1:
             logger.warning("GPT Assistant only supports one OpenAI client. Using the first client in the list.")
+
+        if len(llm_config["config_list"]) > 0:
+            model_name = llm_config["config_list"][0].pop("model", "gpt-4-1106-preview")
```

Comment:
The separation of `model` from other constructors is already handled by `OpenAIWrapper` in its constructor.
It's OK for this PR.

Reply:
Ditto. We can start with a workaround for now, and I'll look into a more unified approach later on.